nservicebus/messaging/routing.md

```diff
@@ -0,0 +1,165 @@
+
+## Send, Publish and Reply
+
+NServiceBus offers three top-level API calls for sending messages, namely `Send`, `Publish` and `Reply`. Each of these calls can have different behavior in terms of message routing. 
+
+## Types of routing
+
+There are two types of routing recognized by NServiceBus, unicast routing and multicast routing.
+
+### Unicast routing
+
+Unicast routing uses queues which are inherently point-to-point channels. Each endpoint instance has an input queue. When sending a message to multiple destinations, a message is copied in-memory at sender endpoint and a copy of the message is dispatched to each destination's input queue.
+
+### Multicast routing
+
+Multicast routing uses topics. Multiple receivers can subscribe to a single topic and each message published to the topic is copied by the messaging infrastructure and delivered to each interested receiver.
+
+### Transport
+
+The transport selected by the user via
+
+```
+config.UseTransport<T>();
+```
+
+decides, for each of the API calls (`Send`, `Publish`, `Reply`), which routing type is used. For example, MSMQ transport uses the unicast routing for all the APIs while the RabbitMQ transport uses the unicast routing for `Send` and `Reply` but multicast routing for `Publish`. 
+
+## Unicast routing
+
+Unicast routing in NServiceBus has a lax layered architecture
+
+|-----------------------------|
+| Type mapping                |
+|-------------|               |
+      |       |               |
+|----------|  |               |
+| Endpoint |  |               |
+| mapping  |  |               |
+|----------|  |---------|     |
+      |                 |     |
+|--------------------|  |     |
+| Instance mapping   |  |     |
+|--------------------|  |-----|
+
+### Type mapping layer
+
+In an ideal world the topmost layer of unicast routing would decide, given a message type, which endpoints should receive the message. In practice such an ideal model is not possible, mostly due to wire-compatibility problems it would cause. A more practical but less strict approach is used where a number of user- and system- defined rules is applied to generate a collection of destinations for a given message type. Each rule operates independently and the result is a logical sum of results of individual rules. Destinations can be expressed either via endpoint name, endpoint instance name or directly via transport address. In last two cases the second and/or third mapping layer is skipped.
+
+#### Dynamic routes
+
+To add a dynamic route you need to pass a function that takes a `Type` and a `ContextBag` containing the context of current message processing and returns a collection of destinations. The function is called each time any message is being send so you need to take performance into account (e.g. use caching if the destinations are to be fetched from a database).
```

Comment:
By looking at configuration, both would need to be defined on busConfiguration and it's not clear what's the use case for dynamic routing? Would an example help?